pkg/epp/plugins/typedname.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugins
```

Comment:
nit: 
this looks and feels much like k8s NamespacedName, so I suggest to follow the same implementation. see here:
https://github.com/kubernetes/apimachinery/blob/master/pkg/types/namespacedname.go

more concretely, I think we can just put Type and Name as public fields, no need for all the functions. 
the only function I would add which is currently missing is the String function which might be useful in logging (you can follow the same separator as NamespacedName).

Reply:
my intention was to replace `Type()` and `Name()` functions with just one function `TypedName()`, since I expect those should always come together when logging or getting the TypedName of a plugin.
so I thought about having a single function in the interface:
```go
func TypeName() types.TypedName
``` 

additionally, in taking inspiration from `NamespacedName` I meant that you could represent it as a string:
`Type/Name (or Name/Type)` which is identical to `NamespacedName`, and if `Name` is not defined, the string would be just `Type`. 

anyway, that's just a nitpick, so please feel free to proceed with your current PR. this is definitely not a blocker.